notebooks/regression-exercises.ipynb

```diff
@@ -0,0 +1,951 @@
+    "categoricalColumns = df_pd.select_dtypes(include=[object]).columns\n",
+    "\n",
+    "print(\"Categorical columns : \" )\n",
+    "print(categoricalColumns)\n",
+    "\n",
+    "impute_categorical = SimpleImputer(strategy=\"most_frequent\")\n",
+    "onehot_categorical =  OneHotEncoder(handle_unknown='ignore')\n",
+    "\n",
+    "categorical_transformer = Pipeline(steps=[('impute',impute_categorical),('onehot',onehot_categorical)])"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "The numerical columns from the data set are identified, and StandardScaler is applied to each of the columns. This way, each value is subtracted with the mean of its column and divided by its standard deviation."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "# Defining the numerical columns \n",
+    "numericalColumns = [col for col in df_pd.select_dtypes(include=[float,int]).columns if col not in ['SALEPRICE']]\n",
+    "print(\"Numerical columns : \" )\n",
+    "print(numericalColumns)\n",
+    "\n",
+    "scaler_numerical = StandardScaler()\n",
+    "\n",
+    "numerical_transformer = Pipeline(steps=[('scale',scaler_numerical)])\n"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "As discussed previously, each of the techniques are grouped by the columns they needed to be applied on and are queued using the ColumnTransformer. Ideally, this is run in the pipeline just before the model is trained. However, to understand what the data will look like, we have transformed the data into a temporary variable."
+   ],
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "source": [
+    "preprocessorForCategoricalColumns = ColumnTransformer(transformers=[('cat', categorical_transformer, categoricalColumns)],\n",
+    "                                                      remainder=\"passthrough\")\n",
+    "preprocessorForAllColumns = ColumnTransformer(transformers=[('cat', categorical_transformer, categoricalColumns),('num',numerical_transformer,numericalColumns)],\n",
+    "                                              remainder=\"passthrough\")\n",
+    "\n",
+    "\n",
+    "#. The transformation happens in the pipeline. Temporarily done here to show what intermediate value looks like\n",
+    "df_pd_temp = preprocessorForCategoricalColumns.fit_transform(df_pd)\n",
+    "print(\"Data after transforming :\")\n",
+    "print(df_pd_temp)\n",
+    "\n",
+    "df_pd_temp_2 = preprocessorForAllColumns.fit_transform(df_pd)\n",
+    "print(\"Data after transforming :\")\n",
+    "print(df_pd_temp_2)"
+   ],
+   "outputs": [],
+   "metadata": {}
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "These are some of the popular preprocessing steps that are applied on the data sets. You can get more information in Data preprocessing in detail."
```

Comment:
Do we need to link to some document/tutorial that gives the details on Data pre-processing